ferry/includes/utils.py

```diff
@@ -17,35 +21,48 @@ def merge_overlapping(sets):
 
     """
 
-    is_merged = True
+    # deduplicate sets to improve performance
+    sets = set([frozenset(x) for x in sets])
+    sets = list(sets)
+
+    g = networkx.Graph()
+    for sub_set in sets:
+        for edge in combinations(list(sub_set), 2):
+            g.add_edge(*edge)
 
-    while is_merged:
+    merged = networkx.connected_components(g)
```

Comment:
hahaha you really pulled out the big guns of networkx for this

Reply:
This actually speeds it up by a huge factor – using the iterative disjoint-set merging approach that I had earlier took about 20 seconds, but this manages to do it in less than 1

Reply:
Whoa that's insane. I knew networkx was good, but didn't realize it was that good!